File: 50 problemas Prolog/programa19.pl
%% --------------------------------------------------------------
%% Nombre: programa19
%% Autor: Van Pratt González Ricardo Adolfo
%% Description: Este programa rota una lista hacia la izquiera n veces.
%%
%% Corresponde al ejercicio 19
%%
%% Fecha: 17 de octubre de 2024
%% --------------------------------------------------------------

% -------- Código en Python (comentado) --------
% def rotate(lst, n):
%     length = len(lst)
%     n = n % length if length > 0 else 0
%     return lst[n:] + lst[:n]

% if __name__ == "__main__":
%     result = rotate([1, 2, 3, 4, 5], 2)
%     print(result)
% ----------------------------------------------

% -------- Código en Prolog --------------------
% Rota una lista N lugares a la izquierda.
% Utiliza la longitud de la lista para manejar rotaciones mayores que la longitud.
rotate(L, N, R) :- length(L, Len), N1 is N mod Len, split(L, N1, L1, L2), append(L2, L1, R).
% ----------------------------------------------

File: ProFuncional/25 programas erlang/programa19.erl
%% --------------------------------------------------------------
%% Módulo: programa19
%% Autor: Van Pratt González Ricardo Adolfo
%% Description: Este programa toma una lista escrita por el usuario
%% y la invierte, imprimiéndola en pantalla.
%%
%% Corresponde al ejercicio 19
%%
%% Fecha: 05 de septiembre de 2024
%% --------------------------------------------------------------

-module(programa19).
-export([inv_lista/1]).

% Pide una lista de valores [1, 2, 3...], la invierte, y devuelve
inv_lista(Lista) ->
    Atsil = lists:reverse(Lista),
    io:format("Lista invertida: ~p~n", [Atsil]).
